### ISSUE TITLE
Bazel: failing build import when `projectview.bazelproject` is missing

### ISSUE THREAD

**Issue by susliko** [scalameta/metals#6131]
### Describe the bug

Some observations:
1. Trying to import build for an existing large project, which is built by bazel and developed using Intellij plugin, fails with a silent error in logs:
```
2024.02.16 17:35:31 INFO  time: Connected to build server in 5.6s
2024.02.16 17:35:31 INFO  Connected to Build server: bazelbsp v3.1.0
2024.02.16 17:35:38 INFO  Failed to run request with params WorkspaceBuildTargetsResult
org.eclipse.lsp4j.jsonrpc.ResponseErrorException:             java.nio.file.NoSuchFileException: path/to/project/projectview.bazelproject
...
Caused by: java.nio.file.NoSuchFileException: path/to/project/projectview.bazelproject/projectview.bazelproject
...
        at org.jetbrains.bsp.bazel.projectview.parser.ProjectViewParser$DefaultImpls.parse(ProjectViewParser.kt:35)
        at org.jetbrains.bsp.bazel.projectview.parser.DefaultProjectViewParser.parse(DefaultProjectViewParser.kt:23)
...
        at scala.meta.internal.metals.RequestMonitorImpl$$anon$1.consume(ServerLivenessMonitor.scala:41)
```

2. `projectview.bazelproject` was automatically generated for an [example](https://github.com/susliko/metals-bazel) project - I can't understand at which moment it happened. Deleteting it breaks build imports and it's not getting generated again even after removing `.bazelbsp`, `.bsp`, `.metals` directories.

### Expected behavior

Generally `projectview.bazelproject` is not something that is present in each bazel project. I'd like to understand how to import a project without this file. Ideally, it should be generated automatically if it's presence is essential for `bazel-bsp` to work

### Operating system

macOS

### Editor/Extension

Nvim (nvim-metals)

### Version of Metals

v1.2.2

### Extra context or search terms

bazel

**Comment by tgodzik** [comment]
Thanks for reporting! projectview file is generated when importing the build, the same as manual import as described in  https://scalameta.org/metals/docs/build-tools/bazel#manual-installation

projectview file should be used in `.bsp/bazel.json` file, which is why it's throwing. Did you remove the file manually or it never got created? The currently used bazel bsp in Metals is a snapshot version so I assume the json file was generated separately?

**Comment by susliko** [comment]
I was using only automatic import proposed by Metals in both cases. I think, an error I was getting in the first case on a large project was caused by my previous tinkering with bazel-bsp, or by `.ijwb/.bazelproject`

Anyway, I've repeatet the following procedure on both old and new [sample](https://github.com/susliko/metals-bazel) project:
- `rm -rf .ijwb .bazelbsp .bsp .metals`
- open editor (nvim), accept suggestion to import build
- await successful import ("in and out, twenty minute adventure" for a project with 300+ targets), ensure code navigation is working
- close editor, remove automatically generated `projectview.bazelproject`
- trigger `Import build` metals command, observe that projectview-file is not getting generated this time. And logs say metals can't find a build target and using presentation compiler:
```
2024.02.16 22:45:24 INFO  time: Imported build in 0.38s
2024.02.16 22:45:24 INFO  Correctly reloaded workspace
2024.02.16 22:45:25 INFO  time: indexed workspace in 1.15s
2024.02.16 22:45:25 WARN  no build target for: /path/to/project/.metals/.tmp/Main2343977158930807829.scala
2024.02.16 22:45:28 INFO  no build target found for /path/to/project/bzlscala/gogo/src/main/scala/gogo/Gogo.scala. Using presentation compiler with project's scala-libra
```

So, this issue narrows down to the following: "projectview.bazelproject should be generated any time build import is triggered and this file is missing or empty"

**Comment by tgodzik** [comment]
Doing a partial fix in #6143 but it seems it's bazel BSP that creates the empty file, which I am not sure it should. Will take a look at it at some point
